From: Jordan Justen <jljusten@debian.org>
Date: Wed, 11 Oct 2023 02:08:57 -0700
X-Dgit-Generated: 2.18.2+metall-0 f3f84e9288151a788c2a7d6ce21a17cf27ec8bd1
Subject: nix: Add various files to clean-files

Signed-off-by: Jordan Justen <jljusten@debian.org>

---

diff --git a/src/nix/local.mk b/src/nix/local.mk
index 20ea29d10..70eaeffb8 100644
--- a/src/nix/local.mk
+++ b/src/nix/local.mk
@@ -27,14 +27,24 @@ $(eval $(call install-symlink, $(bindir)/nix, $(libexecdir)/nix/build-remote))
 
 src/nix-env/user-env.cc: src/nix-env/buildenv.nix.gen.hh
 
+clean-files += src/nix-env/buildenv.nix.gen.hh
+
 src/nix/develop.cc: src/nix/get-env.sh.gen.hh
 
+clean-files += src/nix/get-env.sh.gen.hh
+
 src/nix-channel/nix-channel.cc: src/nix-channel/unpack-channel.nix.gen.hh
 
+clean-files += src/nix-channel/unpack-channel.nix.gen.hh
+
 src/nix/main.cc: doc/manual/generate-manpage.nix.gen.hh doc/manual/utils.nix.gen.hh
 
+clean-files += doc/manual/generate-manpage.nix.gen.hh doc/manual/utils.nix.gen.hh
+
 src/nix/doc/files/%.md: doc/manual/src/command-ref/files/%.md
 	@mkdir -p $$(dirname $@)
 	@cp $< $@
 
 src/nix/profile.cc: src/nix/profile.md src/nix/doc/files/profiles.md.gen.hh
+
+clean-files += src/nix/doc/files/profiles.md.gen.hh
